src/engine/strat_engine/pool.rs

```diff
@@ -219,16 +234,28 @@ impl StratPool {
         cachedevs: Vec<StratBlockDev>,
         timestamp: DateTime<Utc>,
         metadata: &PoolSave,
-        action_avail: ActionAvailability,
+        encryption_info: Option<PoolEncryptionInfo>,
     ) -> BDARecordResult<(Name, StratPool)> {
         if let Err(e) = check_metadata(metadata) {
             return Err((e, tiers_to_bdas(datadevs, cachedevs, None)));
         }
 
         let mut backstore =
             Backstore::setup(uuid, &metadata.backstore, datadevs, cachedevs, timestamp)?;
+        let action_avail = max(
+            get_pool_state(encryption_info),
+            backstore.action_availability(),
+        );
```

Comment:
I think our original intention with `get_pool_state()` was to have a single place where the action availability was determined. We may want to pass a reference to the backstore into `get_pool_state()` to unify the logic of determining the action availability.

Reply:
Let me think about that one...